### ISSUE TITLE
feat req: in-app version downgrade

### ISSUE THREAD

**Issue by alberto-m-santos** [yuliskov/SmartTube#3649]
In the settings menu, possibility to downgrade to the previous beta version or to select one specific version from the list

**Comment by procobain** [comment]
It was YouTube that changed how things work, almost all alternative YouTube programs stopped working.

**Comment by alberto-m-santos** [comment]
Yeah, I didn't mean specifically regarding all the 403 that have been reported. I just thought it would be a good thing to have (though being also aware that beta releases are ofc betas)

**Comment by TecnikOfficial** [comment]
> It was YouTube that changed how things work, almost all alternative YouTube programs stopped working.

REVANCED is working

**Comment by recrof** [comment]
> > It was YouTube that changed how things work, almost all alternative YouTube programs stopped working.
> 
> REVANCED is working welp there is not revanced for tv version gonna wait for fix ... i rather wait then switching to ad infested official youtube app i m using tv browser to watch youtube on tv ad free

revanced is de-facto official app that has been patched - no wonder it's working fine.

**Comment by lothkang** [comment]
> > It was YouTube that changed how things work, almost all alternative YouTube programs stopped working.
> 
> REVANCED is working welp there is not revanced for tv version gonna wait for fix ... i rather wait then switching to ad infested official youtube app i m using tv browser to watch youtube on tv ad free

I wrote a quick tutorial on using ReVanced on an Android TV. All you really need is a cheapo wireless mouse plugged into a free USB port or a phone with a free Bluetooth keyboard/mouse app (e.g. BT Remote) and a launcher, like FLauncher to access sideloaded apps. But 22.38 is working, so you don't need ReVanced.

**Comment by Kougami901** [comment]
>All you really need is a cheapo wireless mouse plugged into a free USB port or a phone with a free Bluetooth keyboard/mouse app (e.g. BT Remote) and a launcher, like FLauncher to access sideloaded apps. But 22.38 is working, so you don't need ReVanced.

Just in case YT will roll out more STN-breaking changes in the future: This is already overkill.
Just use the built in chromecast/miracast of your android stb and stream revanced from your phone. Is it as convenient as a native app? No. Does it work, when YT tries to break things? Yes.

**Comment by notamaiar** [comment]
> > It was YouTube that changed how things work, almost all alternative YouTube programs stopped working.
> 
> REVANCED is working welp there is not revanced for tv version gonna wait for fix ... i rather wait then switching to ad infested official youtube app i m using tv browser to watch youtube on tv ad free

Clipious is still working. Not as full-featured as Smarttube, but a serviceable backup.

**Comment by TecnikOfficial** [comment]
> Clipious is still working. Not as full-featured as Smarttube, but a serviceable backup.

thank you for recommending [clipious](https://github.com/lamarios/clipious) is a useful backup just in case only needed voice option in search in clipious app
